Feat:[Authentication system]

Evaristekanova opened the pull request:
### What does this PR do?

- add session controller(login functionality)
- add authenticable module

Evaristekanova committed:
# This is a combination of 4 commits.
# This is the 1st commit message:

ignore some files

# This is the commit message #2:

set up active admin
setup authentication
customize active admin

# This is the commit message #3:

implement authorization system
add authenticable module

protect some of route

# This is the commit message #4:

modify authenticable module
set up SMTP settings for sending email
set up dotenv
protect some route

Evaristekanova committed:
implement authorization system
add authenticable module

protect some of route

benessilfie committed:
Update sessions_controller.rb

benessilfie committed:
Update users_controller.rb

benessilfie committed:
Update user.rb

benessilfie commented on app/controllers/concerns/authenticable.rb:
Does this work? I dont see you extracting the token. you just passed the authorization hash to the decoded function.

Evaristekanova committed:
set up active admin
setup authentication
customize active admin

implement authorization system
add authenticable module

protect some of route

modify authenticable module
set up SMTP settings for sending email
set up dotenv
protect some route

Update sessions_controller.rb

Update users_controller.rb

Update user.rb

Evaristekanova committed:
rebase to match remote and local
